src/lib/StrategyLib.sol

```diff
@@ -72,3 +80,36 @@ function computeDeltaYGivenDeltaL(
 ) pure returns (uint256 deltaX) {
     return reserveY.mulDivUp(deltaL, liquidity);
 }
+
+function encodeAllocationDeltasGivenDeltaX(
+    uint256 deltaX,
+    uint256 reserveX,
+    uint256 reserveY,
+    uint256 liquidity
+) pure returns (bytes memory) {
+    uint256 deltaY = computeDeltaYGivenDeltaX(deltaX, reserveX, reserveY);
+    uint256 deltaL = computeDeltaLGivenDeltaX(deltaX, liquidity, reserveX);
+    return abi.encode(deltaX, deltaY, deltaL);
+}
+
+function encodeAllocationDeltasGivenDeltaY(
+    uint256 deltaY,
+    uint256 reserveX,
+    uint256 reserveY,
+    uint256 liquidity
+) pure returns (bytes memory) {
+    uint256 deltaX = computeDeltaXGivenDeltaY(deltaY, reserveX, reserveY);
+    uint256 deltaL = computeDeltaLGivenDeltaY(deltaY, liquidity, reserveY);
+    return abi.encode(deltaX, deltaY, deltaL);
+}
+
+function encodeAllocationDeltasGivenDeltaL(
+    uint256 deltaL,
+    uint256 reserveX,
+    uint256 reserveY,
+    uint256 liquidity
+) pure returns (bytes memory) {
+    uint256 deltaX = computeDeltaXGivenDeltaL(deltaL, reserveX, liquidity);
+    uint256 deltaY = computeDeltaYGivenDeltaL(deltaL, reserveY, liquidity);
+    return abi.encode(deltaX, deltaY, deltaL);
+}
```

Comment:
I'm not sure if these functions should stay here, maybe we can have them in the PairStrategy contract directly.

Reply:
Would moving them change the api at all? If not i think moving them is fine.

Reply:
These should for sure go in PairStrategy, clement is right